File: backend/ReservationApp.Application/DTOs/AdminReservationQueryParams.cs
using ReservationApp.Domain.Enums;

namespace ReservationApp.Application.DTOs;

public class AdminReservationQueryParams
{
    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public DateTime DateFrom { get; set; }
    public DateTime DateTo { get; set; }
    public Guid? RestaurantId { get; set; }
    public string? Department { get; set; }
    public ReservationStatus? Status { get; set; }
    public string? Search { get; set; }
}

File: android/domain/src/main/java/com/reservationapp/domain/model/AdminReservationQueryParams.kt
package com.reservationapp.domain.model

data class AdminReservationQueryParams(
    val page: Int = 1,
    val pageSize: Int = 10,
    val dateFrom: String? = null,
    val dateTo: String? = null,
    val search: String? = null,
    val status: String? = null
)

